core/Container/Container.php

```diff
@@ -32,6 +32,11 @@ public function __construct(
         $this->resolvedEntries[self::class] = $this;
     }
 
+    /**
+     * @template T of object
+     * @param class-string<T>|string $name Container entry name.
```

Comment:
Should we add this too?

```suggestion
     * @param class-string<T>|string $name Container entry name.
     * @throws NotFoundException
```

Reply:
That should be "auto-detectable"

Reply:
I’m not sure to understand what you mean here. There is no need to precise it since it’s auto-detectable?

Reply:
I guess that's something we need to decide in general. I'm currently only trying to add type hints for those variables, where the code doesn't or can't define the explicitly. We can for sure consider to add doc blocks that contain everything, but that will also cause more maintaining work to keep everything up to date